src/include/function/gds/gds_frontier.h

```diff
@@ -147,6 +140,67 @@ class GDSFrontier {
     common::table_id_map_t<common::offset_t> numNodesMap;
 };
 
+/*
+ * LocalFrontierSample keeps a small frontier per thread.
+ * If enabled, this frontier represents a complete local frontier.
+ * Otherwise, complete local frontier is larger than this sample.
+ * */
+class LocalFrontierSample {
```

Comment:
Reading the code, this looks redundant to me. It looks like a specialized SparseFrontier implementation that keeps a set of nodeIDs instead of a set of set of nodeOffsets. It looks like we should be able to have a single SparseFrontier implementation that can be used both as a LocalFrontier and a global one, even if it is a slightly slower implementation.

Reply:
Done